src/org/zowe/jenkins_shared_library/pipelines/generic/GenericPipeline.groovy

```diff
@@ -1519,6 +1501,22 @@ class GenericPipeline extends Pipeline {
         }
     }
 
+    Boolean checkAuthorizedUser() {
+        def isAuthorizedUser = false
+        Integer causeID = this.build.identifySurfaceCause()
+        if (causeID == PipelineConstants.UPSTREAM_CAUSE_ID) {
+            causeID = this.build.identifyRootCause()
+        }
+        if (causeID == PipelineConstants.USERID_CAUSE_ID) {
```

Comment:
We also need to handle `TIMER_CAUSE_ID`. It should be this cause for nightly promoting pipeline. If a nightly promoting pipeline triggers a zowe-install-packaging or zowe-install-test job, I believe the root cause should be `TIMER_CAUSE_ID`. I couldn't think of a good way to determine if the scheduled job is good or bad, so let's just allow it for now.

Reply:
For nightly timer, my current plan is to not allow it, because it will run convenience build, which duplicates what nightly build is supposed to do.